### PR TITLE
Fix #804 - Fixes Wrong Fragment selection

### LINKED ISSUE #804: Wrong Fragment selection in Navigation Pane
**Summary:** 

The selection in Navigation pane remains as such if we select an option which creats a new activity(such as Settings, Path Tracker,or Collection Sheet), and the navigate back using back button.

**Steps to reproduce:** 

1. Open Settings, Path Tracker or Collection Sheet from navigation pane.
2. Press back button.

**Expected behavior:** 

The selection in navigation pane should be according to the previously opened option(fragment or activity).

**Observed behavior:** 

The selection in navigation pane is according to the new option opened, the one from which we navigated back.

**Device and Android version:** 

Model: Samsung Galaxy A5(2016) 
Android version: Android 7.0 Nougat. 

**Screenshots:** 

ezgif com-video-to-gif

**Would you like to work on the issue?**

Yes.

### PR DESCRIPTION
**Pull request by OussEmaDevCode**
Fixes #804 

Please Add Screenshots If there are any UI changes.
video2gif_20191212_115515

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by OussEmaDevCode** [comment]
@luckyman20 @moksh-mahajan Hi, It's been more than a day since I've pushed this PR please have a look !!!

**Comment by moksh-mahajan** [comment]
@OussEmaDevCode Sorry for the delay. 
On it.

**Comment by manigedit** [comment]
@moksh-mahajan I hope  I am not needed to intervene.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java [hunk 1]
```diff
@@ -237,6 +237,22 @@ public boolean onNavigationItemSelected(MenuItem item) {
         return true;
     }
 
+    @Override
+    protected void onResume() {
+        super.onResume();
+        Fragment currentFragment = getSupportFragmentManager().findFragmentById(R.id.container);
+        if (currentFragment instanceof SearchFragment) {
```
Comment (luckyman20) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java]: Use switch instead.

**Comment by OussEmaDevCode** [comment]
@moksh-mahajan @luckyman20 Hi, It's simply impossible to use switch statement with instanceof operator because the switch statement can only contain case statements which are compile time constants and which evaluate to an integer up to Java 6 and a string in Java 7. And Since I didn't want to deceive you I did this solution with involves the class names instead of the actual classes. Hope it's fine !!!!

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java [hunk 2]
```diff
@@ -237,6 +237,30 @@ public boolean onNavigationItemSelected(MenuItem item) {
         return true;
     }
 
+    @Override
+    protected void onResume() {
+        super.onResume();
+        String currentFragment = getSupportFragmentManager().findFragmentById(R.id.container)
+                .getClass()
+                .getSimpleName();
```
Comment (moksh-mahajan) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java]: This can be included in the previous line.

Comment (OussEmaDevCode) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java]: That will create a style error like this 'the line exceeds 100 characters'

**Comment by OussEmaDevCode** [comment]
@moksh-mahajan That will create a style error like this 'the line exceeds 100 characters'

**Comment by moksh-mahajan** [comment]
@OussEmaDevCode I meant to say line no. 245 can be merged with line no. 244

**Comment by OussEmaDevCode** [comment]
@moksh-mahajan Done !!!

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java [hunk 3]
```diff
@@ -237,6 +237,29 @@ public boolean onNavigationItemSelected(MenuItem item) {
...
+                break;
+            case "GroupsListFragment":
+                mNavigationView.setCheckedItem(R.id.item_groups);
+                break;
+            case "CenterListFragment":
+                mNavigationView.setCheckedItem(R.id.item_centers);
+                break;
+            case "OfflineDashboardFragment":
+                mNavigationView.setCheckedItem(R.id.item_offline);
+                break;
```
Comment (moksh-mahajan) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java]: No need of break for the last case.

**Comment by OussEmaDevCode** [comment]
@moksh-mahajan Done !!

**Comment by OussEmaDevCode** [comment]
@moksh-mahajan is it going to be merged ?

**Pull request merged**